.github/workflows/build.yml

```
@@ -134,35 +142,51 @@ jobs:
-          # dev_tags are tags that get automatically pushed whenever successful
-          # builds make it to the stable channel. The intention is for these tags
-          # to be used for early testing of new code prior to official releases
-          # going out. The stable channel implies that all tests and scans have
-          # completed successfully, so these images should be _stable_ but are not
-          # intended for production use.
-          #
-          # Here we have two example dev tags. The first (ending -dev) is a tag
-          # that will be updated over-and-over as new builds arrive in stable.
-          #
-          # The second (using the git SHA) will produce a new separate tag for
-          # each commit that is built. (These can still be overridden if the same
-          # commit is built successfully a second time, but that is a less likely
-          # scenario.) These kinds of dev tags are useful if you want to be able
-          # to use Docker images built from those specific commits.
-          #
-          # NOTE: dev_tags MUST publish to the 'hashicorppreview' DockerHub org, it
-          # will fail to any other DockerHub org or registry. You can optionally
-          # prepend docker.io
           dev_tags: |
             hashicorppreview/${{ env.repo }}:${{ env.version }}
-            hashicorppreview/${{ env.repo }}:${{ env.version }}-${{ github.sha }}
+            docker.io/hashicorppreview/${{ env.repo }}:${{ env.version }}-${{ github.sha }}
+
+  build-docker-fips:
+    name: Docker ${{ matrix.arch }} FIPS release build
+    needs:
+      - get-product-version
+      - build-linux
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - { arch: "arm64" }
+          - { arch: "amd64" }
+    env:
+      repo: ${{ github.event.repository.name }}
+      version: ${{ needs.get-product-version.outputs.product-version }}+fips1402
+
+    steps:
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - name: Docker FIPS Build (Action)
+        uses: hashicorp/actions-docker-build@v1
+        with:
+          smoke_test: |
+            TEST_VERSION="$(docker run "${IMAGE_NAME}" version | awk '{print $2}')"
+            echo "TEST_VERSION=${TEST_VERSION}"
+            echo "version=${version}"
+            if [ "${TEST_VERSION}" != "v${version}" ]; then
+              echo "Test FAILED"
+              exit 1
+            fi
+            echo "Test PASSED"
+          version: ${{ env.version }}
+          target: release-fips-default
+          arch: ${{ matrix.arch }}
+          tags: |
+            docker.io/hashicorp/${{ env.repo }}-fips:${{ env.version }}
+            public.ecr.aws/hashicorp/${{ env.repo }}-fips:${{ env.version }}
+          dev_tags: |
+            hashicorppreview/${{ env.repo }}-fips:${{ env.version }}
+            docker.io/hashicorppreview/${{ env.repo }}-fips:${{ env.version }}${{ github.sha }}
```

I think we're missing a `-` between the version and the SHA:

```suggestion
            docker.io/hashicorppreview/${{ env.repo }}-fips:${{ env.version }}-${{ github.sha }}
```